src/IW/Db/Repo.hs

```diff
@@ -43,3 +45,14 @@ upsertRepos = executeMany [sql|
         descr = EXCLUDED.descr
       , categories = EXCLUDED.categories;
 |]
+
+-- | Update a repo's categories field.
+updateRepoCategories :: (MonadCabal m, WithDb env m) => Repo -> m ()
+updateRepoCategories Repo{..} = do
+    categories <- getCabalCategories repoOwner repoName
+    execute [sql|
```

Comment:
A good practice that we learned during development is that it's usually a good idea to put SQL queries in separate functions and not interleave them with other effects. That leads to simpler code and helps with testing. For example, if you want to test `updateRepoCategories` function, you need to pass the whole repo, and the function will download something. So for the tests for this function to work, you actually need to give valid testing data connected to the real world. Also, there is no much benefit in testing `getCabalCategories + SQL` query because we already have tests for `getCabalCategories` so need to test only SQL queries.

Since we're writing raw SQL in Haskell code, we need to test every our query (that it doesn't contain syntax errors, that it works as we expect). So we can make our lives easier by decomposing all SQL queries into separate functions. The most difficult functions to test in our codebase are the functions that do multiple actions of different effects inside them. Haskell allows to decompose problems into smaller pieces, test them independently and combine with confidence. We need to write only a couple of tests for `getCabalCategories`, and we will be sure that the function works. Same for SQL queries. And then we don't need to write tests for functions that just compose multiple tested functions (not always true if logic is not trivial, but this is what we're trying to achieve ultimately).

Reply:
@chshersh Understood. Makes sense :bulb: